src/json/json.cr

```diff
@@ -1,3 +1,40 @@
+# JavaScript Object Notation (JSON)
+#
+# JSON (JavaScript Object Notation) is a way to write data in Javascript.
+# Like XML, it allows to encode structured data in a text format that can
+# be easily read by humans Its simple syntax and native compatibility with
+# JavaScript have made it a widely used format.
+#
+# An object is a series of string keys mapping to values, in `"key": value`
+# format. Arrays are enclosed in square brackets `[ ... ]` and objects in
+# curly brackets `{ ... }`.
+#
+# Crystal provides a mechanism for encoding & decoding of values to and
+# from JSON via the serialization API.
+#
+# To read more about JSON visit: json.org
+#
+# Parsing JSON
+#
+# To parse a JSON string received by another application or generated within
+# your existing application:
+#
+# ```
+# require "json"
+#
+# hash = JSON.parse("{\"hello\": \"goodbye\"}") as Hash
+# hash["hello"] # => {"hello": "goodbye"}
+# ```
+#
+# Generating JSON
+#
+# Creating a JSON string for communication or serialization is just as simple.
+#
+# ```
+# require "json"
+#
+# { hello: "goodbye" }.to_json # => "{\"hello\":\"goodbye\"}"
+# ```
```

Comment:
I feel like we should also reference `json_mapping` here and maybe even say it's the recommended way to parse JSON with Crystal. See https://github.com/manastech/crystal/issues/982#issuecomment-121156428